server/src/domain/plugin_backend_handler.rs

```diff
@@ -0,0 +1,510 @@
+            plugin_handler: self.plugin_handler.clone(),
+            request_context: context,
+        }
+    }
+}
+
+#[async_trait]
+impl ReadSchemaBackendHandler for PluginBackendHandler {
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn get_schema(&self) -> Result<Schema> {
+        self.backend_handler.get_schema().await
+    }
+}
+
+#[async_trait]
+impl SchemaBackendHandler for PluginBackendHandler {
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn add_user_attribute(&self, request: CreateAttributeRequest) -> Result<()> {
+        self.backend_handler.add_user_attribute(request).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn add_group_attribute(&self, request: CreateAttributeRequest) -> Result<()> {
+        self.backend_handler.add_group_attribute(request).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn delete_user_attribute(&self, name: &AttributeName) -> Result<()> {
+        self.backend_handler.delete_user_attribute(name).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn delete_group_attribute(&self, name: &AttributeName) -> Result<()> {
+        self.backend_handler.delete_group_attribute(name).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn add_user_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.backend_handler.add_user_object_class(name).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn add_group_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.backend_handler.add_group_object_class(name).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn delete_user_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.backend_handler.delete_user_object_class(name).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn delete_group_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.backend_handler.delete_group_object_class(name).await
+    }
+}
+
+#[async_trait]
+impl GroupBackendHandler for PluginBackendHandler {
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn get_group_details(&self, group_id: GroupId) -> Result<GroupDetails> {
+        self.backend_handler.get_group_details(group_id).await
+    }
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn update_group(&self, request: UpdateGroupRequest) -> Result<()> {
+        let ctx = self.new_plugin_context();
+        let request = self
+            .plugin_handler
+            .on_update_group(ctx.clone(), request.clone())
+            .await
+            .unwrap_or(request);
+        self.backend_handler.update_group(request.clone()).await?;
+        let _ = self.plugin_handler.on_updated_group(ctx, request).await;
+        Ok(())
```

Comment:
We can probably extract that to a function, or at least a macro

Reply:
Not a bad idea with a macro. Were you thinking the whole block?

Reply:
Yeah, reduce boilerplate to enforce consistency. Something like
```rust
wrap_call_with_plugin_callbacks!(update_group, on_update_group, on_updated_group);
```

Reply:
Finished the last few handlers. Done.